fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/JsonInsert.java

```diff
@@ -47,6 +44,21 @@ public JsonInsert(Expression arg0, Expression arg1, Expression arg2, Expression.
         super("json_insert", ExpressionUtils.mergeArguments(arg0, arg1, arg2, varArgs));
     }
 
+    @Override
+    public FunctionSignature customSignature() {
+        List<DataType> arguments = new ArrayList<>();
+        arguments.add(VarcharType.SYSTEM_DEFAULT); // json_str
+        for (int i = 1; i < arity(); i++) {
+            if ((i & 1) == 0 && (getArgumentType(i).isComplexType() || getArgumentType(i).isJsonType())) {
```

Comment:
what does `(i & 1) == 0` mean?

Reply:
(i & 1) == 0 determines the even position, which exactly corresponds to the position of the value parameter. pass the origin value type to BE and should not cast to varchar